spec/lib/workflow/stores/mongo-spec.js

```diff
@@ -592,6 +599,51 @@ describe('Task Graph mongo store interface', function () {
         });
     });
 
+    it('updates dependent tasks under anyOf key', function() {
+
+        var waterlineUnderAnyOf = {
+            taskdependencies:{
+                find: sinon.stub().returns([]).resolves(),
```

Comment:
The `returns` and `resolves` will conflict, the `returns` output a non-promise result, the `resolves` output a promise result. Since  `taskdependencies.find` should return promise, so keep `resolves()` is enough, or change to `resolves([])`